src/transformers/models/llava_next_video/modeling_llava_next_video.py

```diff
@@ -910,7 +910,7 @@ def forward(
             pass
 
         # generation with cache, decoding stage
-        elif past_key_values is not None and (pixel_values is not None or pixel_values_videos is not None):
+        elif pixel_values is not None or pixel_values_videos is not None:
```

Comment:
Not sure why this is changed as the first if is ` inputs_embeds.shape[1] != 1:` so that is prefill / simple forward (and indeed merging should always be done regardless of generation) 
But then here, you need to index the past key values, which don't always exist (forward call with use_cache = False). for example

Reply:
In forward call with no cache, we should go with the first 'if' and never reach the `elif`. yeah kinda too many dependencies, but shouldn't cause an error. I'll add a parametrized to the new tests (use-cache or no-cache)